spring-integration-core/src/main/java/org/springframework/integration/endpoint/SourcePollingChannelAdapter.java

```diff
@@ -214,8 +218,22 @@ protected void handleMessage(Message<?> message) {
 		}
 		try {
 			this.messagingTemplate.send(getOutputChannel(), message);
+			if (this.source instanceof DeferredAcknowlegmentMessageSource) {
```

Comment:
Why do we need this new abstraction when there really would be enough to get an `IntegrationMessageHeaderAccessor.ACKNOWLEDGMENT_CALLBACK` header from the message and perform the logic if it isn't `null`?

Reply:
Makes sense.